src/animation/AnimationUtils.d.ts

```diff
@@ -23,4 +23,11 @@ export namespace AnimationUtils {
 		endFrame: number,
 		fps?: number
 	): AnimationClip;
+	export function makeClipAdditive(
+		sourceClip: AnimationClip,
+		referenceFrame?: number,
+		cloneOriginal?: boolean,
+		clonedName?: string,
+		fps?: number
+	): AnimationClip;
```

Comment:
Could you help me understand how `makeClipAdditive` should be used? I am new to animation blending modes, and trying to learn this. :) From reading a bit, it sounds like one common workflow would involve starting with several standalone — not yet additive — animations:

- idle
- walk
- run
- arm wave `(~‾⌣‾ )~`

Now suppose we want to make the "arm wave" animation additive, so it can be played on top of the others. Is it always the case that the additive animation must have a frame at the base pose? Might the artist need to make it additive by subtracting a different animation, or a static pose, from it? 

I guess I am wondering if the `makeClipAdditive` function should perhaps take _two_ clips, `referenceClip` and `targetClip`, and subtract the reference frame of `referenceClip` from each frame of `targetClip`? Like:

```ts
makeClipAdditive(referenceClip, referenceFrame, targetClip, fps = 30)
```

Reply:
makeClipAdditive would be generally be used for animations on hierarchies, like a biped skeleton. Additive animations add on top of the result of non-additive animations, so if you want the character's 'bones' to remain the same length when you add an additive animation, you'd need to author the animation with all of the joints having local translation values of [0,0,0]. Not very intuitive since you just end up with a blob at the origin. Instead you can author the animation as you normally would and then use makeClipAdditive to subtract a pose out, which would result in zeroing out the translation values if you did not change them during the animation.

I like your idea of including a reference clip as an input, seems like it will make things a bit more flexible. I will definitely add that in, thanks for the feedback!

Reply:
> ... you'd need to author the animation with all of the joints having local translation values of [0,0,0]. Not very intuitive since you just end up with a blob at the origin. Instead you can author the animation as you normally would ...

I see, thanks — it appears that authoring programs like Maya and Blender do support additive animation. Do you mean that creating the animation that way is just less convenient or less common?

Reply:
Those programs do support additive animation, but it's not a standard feature for exported formats. The only exported file format I know of that supports additive animation is FBX, and it requires that the  animation layers be kept intact. This is not a common export case because most people animate using control rigs and so have to bake the results when exporting. Baking merges everything into the base layer and results in adding the reference pose/animation back in.